src/core/public/http/external_url_service.ts

```diff
@@ -50,20 +50,33 @@ function normalizeProtocol(protocol: string) {
   return protocol.endsWith(':') ? protocol.slice(0, -1).toLowerCase() : protocol.toLowerCase();
 }
 
+const createIsInternalUrlValidation = (
+  location: Pick<Location, 'href'>,
+  serverBasePath: string
+) => {
+  const base = new URL(location.href);
+  return function isInternallUrl(next: string) {
+    const url = new URL(next, base);
+
+    return (
+      url.origin === base.origin &&
+      (!serverBasePath || url.pathname.startsWith(`${serverBasePath}/`))
+    );
+  };
+};
+
 const createExternalUrlValidation = (
   rules: IExternalUrlPolicy[],
   location: Pick<Location, 'href'>,
   serverBasePath: string
 ) => {
+  const base = new URL(location.href);
```

Comment:
This won't work for all cases - it's important the `base` URL is built out of `location.href` at the very moment the validation takes place. This is the case because relative paths need to be resolved against the current state. I just recently fixed this problem (see here: https://github.com/elastic/kibana/pull/116404).

Please make sure this is happening for both the new helper and the existing `validateExternalUrl` function

Reply:
done